portfolio/src/main/java/com/google/sps/servlets/DeleteContactServlet.java

```
@@ -0,0 +1,37 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import static java.lang.System.out;
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/delete-data")
+public class DeleteContactServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Email");
```

Now that you're using the "Email" string in a few places, it might be a good idea to define it as a constant in a shared location

---

I was actually thinking about this but I only use the string "Email" twice so didn't feel it was necessary, Let me know if you disagree.

---

I think it can't hurt anyway.  Typically, if we use a hard-coded value even once, well use a constant for it.  It helps prevent errors if you do start using it again, and will make the code more readable

---

so where would you recommend i declare the public static final String = "Email"?

---

IMO what i would do is have a separate file called DataStoreKeys or similar and include it as a public filed there. You could also bring the other keys you are using for the datastore into that file as well (timestamp, any others as you keep going)